tests/hotplug/memory.go

```diff
@@ -0,0 +1,146 @@
+
+			vmi := libvmi.NewAlpineWithTestTooling(
+				libvmi.WithMasqueradeNetworking()...,
+			)
+			vmi.Namespace = testsuite.GetTestNamespace(vmi)
+			vmi.Spec.Domain.Memory = &v1.Memory{
+				Guest: &guest,
+			}
+			vm := tests.NewRandomVirtualMachine(vmi, true)
+			vm.Spec.LiveUpdateFeatures = &v1.LiveUpdateFeatures{
+				Memory: &v1.LiveUpdateMemory{
+					MaxGuest: &maxGuest,
+				},
+			}
+
+			vm, err := virtClient.VirtualMachine(vm.Namespace).Create(context.Background(), vm)
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(ThisVM(vm), 360*time.Second, 1*time.Second).Should(beReady())
+			libwait.WaitForSuccessfulVMIStart(vmi)
+
+			By("Ensuring the compute container has at least 128Mi of memory")
+			compute := tests.GetComputeContainerOfPod(tests.GetVmiPod(virtClient, vmi))
+
+			Expect(compute).NotTo(BeNil(), "failed to find compute container")
+			reqMemory := compute.Resources.Requests.Memory().Value()
+			Expect(reqMemory).To(BeNumerically(">=", guest.Value()))
+
+			By("Hotplug 128Mi of memory")
+			patchData, err := patch.GenerateTestReplacePatch("/spec/template/spec/domain/memory/guest", "128Mi", "256Mi")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, patchData, &k8smetav1.PatchOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for HotMemoryChange condition to appear")
+			Eventually(ThisVMI(vmi), 1*time.Minute, 2*time.Second).Should(HaveConditionTrue(v1.VirtualMachineInstanceMemoryChange))
+
+			By("Making sure that a parallel memory change is not allowed during an ongoing hotplug")
+			patchData, err = patch.GenerateTestReplacePatch("/spec/template/spec/domain/memory/guest", "256Mi", "128Mi")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, patchData, &k8smetav1.PatchOptions{})
+			Expect(err).To(HaveOccurred())
+
+			By("Ensuring live-migration started")
+			var migration *v1.VirtualMachineInstanceMigration
+			Eventually(func() bool {
+				migrations, err := virtClient.VirtualMachineInstanceMigration(vm.Namespace).List(&k8smetav1.ListOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				for _, mig := range migrations.Items {
+					if mig.Spec.VMIName == vmi.Name {
+						migration = mig.DeepCopy()
+						return true
+					}
+				}
+				return false
+			}, 30*time.Second, time.Second).Should(BeTrue())
+			libmigration.ExpectMigrationToSucceedWithDefaultTimeout(virtClient, migration)
+
+			By("Ensuring the libvirt domain has more available guest memory")
+			Eventually(func() int64 {
+				vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				return getCurrentDomainMemory(vmi).Value()
+			}, 240*time.Second, time.Second).Should(BeNumerically(">", guest.Value()))
```

Comment:
Because with virtio-mem hotplug you're not guaranteed to get all the memory you asked for (instantly), the kernel will try but it's best effort, so if I use strict equality here the test might become flaky.